modules/post/multi/gather/lastpass_creds.rb

```diff
@@ -172,15 +152,18 @@ def find_db_paths(path, browser, account)
     paths = []
 
     vprint_status "Checking #{account}'s #{browser}"
-    if browser == "Firefox" # Special case for Firefox
-      profiles = firefox_profile_files(path, browser)
-      paths |= profiles
+    if browser == "IE" # Special case for IE
+      data = read_registry_key_value('HKEY_CURRENT_USER\Software\LastPass', "LoginUsers")
+      data = read_registry_key_value('HKEY_CURRENT_USER\Software\AppDataLow\Software\LastPass', "LoginUsers") if data.blank?
+      paths |= ['HKEY_CURRENT_USER\Software\AppDataLow\Software\LastPass'] if !data.blank? && path != "Low\\LastPass" # Hacky way to detect if there is access to user's data (attacker has no root access)
+    elsif browser == "Firefox" # Special case for Firefox
+      paths |= firefox_profile_files(path)
     else
-      paths |= file_paths(path, browser, account)
+      paths |= file_paths(path)
     end
 
     vprint_good "Found #{paths.size} #{browser} databases for #{account}"
-    paths
+    return paths.size > 0 ? paths.first : []
```

Comment:
This is a strange pattern, IMO, and doesn't align with what the comments now say.  IMO, it should return a list of paths, and if that list is empty, it is assumed that no paths were found.  If any paths were found, all are returned, but in general only the first is used (at least as implemented here).  That should clean this function and its use up a bit.